nuxeo-services/nuxeo-permissions/src/test/java/org/nuxeo/ecm/permissions/TestPermissionListener.java

```diff
@@ -193,6 +197,25 @@ public void shouldUpdateDirectory() throws Exception {
         }
     }
 
+    @Test
+    public void shouldUpdateDirectoryOnDocumentDelete() {
+        // Create a document with an ACE
+        DocumentModel doc = createTestDocument();
+        ACP acp = doc.getACP();
+        acp.addACE(ACL.LOCAL_ACL, new ACE("fry", "Browse", true));
+        doc.setACP(acp, true);
+
+        // Remove the document
+        session.removeDocument(doc.getRef());
+        txFeature.nextTransaction();
+
+        // Check ACE info is removed
+        try (Session dirSession = directoryService.open(ACE_INFO_DIRECTORY)) {
+            DocumentModelList entries = dirSession.query(Collections.singletonMap("docId", doc.getId()));
```

Comment:
If the operators don't add manually the docId index this will never find anything.
Maybe the listener should be disabled by default ?